engine/src/main/java/org/terasology/rendering/dag/WireframeTask.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.lwjgl.opengl.GL11;
+import static org.lwjgl.opengl.GL11.GL_FILL;
+import static org.lwjgl.opengl.GL11.GL_FRONT_AND_BACK;
+import static org.lwjgl.opengl.GL11.GL_LINE;
+
+/**
+ *
+ */
+public class WireframeTask implements PipelineTask { // TODO: separate this into SetWireframeOnTask and SetWireframeOffTask?
```

Comment:
_After thinking extensively on your comment:_

👍 for not relying on null values. 

First thing that I thought about achieving your suggestion would be to:

```java
class FooNode {
    public void initalise() {
        if (debugRenderingConfig.isWireframe()) {
            addDesiredStateChange(new WireframeStateChange());
        }
    }
    ...
    public void process() {
         if (debugRenderingConfig.isWireframe()) { // :(
            RenderSystem.restart() // or something simillar
        }
    }
}
```
But the problematic part is having bunch of if's inside `Node`'s `process()` method. (6-7 nodes are going to have these.) For eliminating that, what if RenderSystem keeps track of these changes. Instead of checking if this is enabled or not, how about directly restarting the system whenever that config is changed. A node can provide a special class for `RenderSystem` for triggering this `restart`.